packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -884,6 +884,11 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
       _controller!.dispose();
       _controller = null;
     }
+
+    if (widget.focusNode != oldWidget.focusNode) {
+      (oldWidget.focusNode ?? _focusNode)?.removeListener(_handleFocusChanged);
+      (widget.focusNode ?? _focusNode)?.addListener(_handleFocusChanged);
```

Comment:
I guess we weren't swapping these properly before 😁

Reply:
The `_handleFocusChange` listener is added in this PR. We didn't have to remove/add listeners before.